core/org.wso2.carbon.base/src/main/java/org/wso2/carbon/base/CarbonBaseUtils.java

```diff
@@ -126,4 +126,32 @@ public static String getCarbonHome() {
 		}
 		return carbonHome;
 	}
+
+	public static String replaceSystemProperty(String text) {
+		int indexOfStartingChars = -1;
+		int indexOfClosingBrace;
+
+		// The following condition deals with properties.
+		// Properties are specified as ${system.property},
+		// and are assumed to be System properties
+		while (indexOfStartingChars < text.indexOf("${")
+				&& (indexOfStartingChars = text.indexOf("${")) != -1
+				&& (indexOfClosingBrace = text.indexOf('}')) != -1) {
+			String sysProp = text.substring(indexOfStartingChars + 2,
```

Comment:
This can also be handled using a regex pattern which will improve the readability.